### ISSUE TITLE
In-layer normalization techniques for training very deep neural networks

### ISSUE THREAD

**Issue by riron1206** [riron1206/read_paper#20]
https://theaisummer.com/normalization/
DNNの正規化手法をまとめた記事

-------------------------------------------------------
■Batch normalization(2015): 1ミニバッチ内の全データの同一チャネルが平均0,分散1に正規化するレイヤー。統計値(平均・標準偏差)が学習されるパラメータ
- 学習早めてくれる
- 統計値により正則化
- BNは各レイヤーの入力分布の変化(内部共変量シフト:Internal Covariate Shift)を低減できると考えていたがそうではなかった
    - 参考: https://gradientscience.org/batchnorm/
- BNには十分に大きなバッチサイズが必要。バッチサイス少ないと統計値にならない
- バッチサイズの変動するときは不向き。学習時の統計値で最適化されてるので学習の時とバッチサイズを変えるとよくない
- dropoutとBNの組み合わせた方が精度出るかもしれないらしい（dropoutない方良い良いと言ってた論文もあったはずだが）

<br>

■Synchronized Batch Normalization(2018): マルチGPUでのBN
普通のBNでData Parallelを使った場合、GPU別にBNが計算されてしまいミニバッチ内のデータの分布が変わってしまう可能性がある
平均・標準偏差といったパラメーターがデバイス間で共通のものとして計算されているため、並列化による分布の変動がおきない
- 参考: https://blog.shikoan.com/sync-batch-norm-pytorch/

<br>

■Layer normalization(2016): 1つのサンプルについて全チャンネルにまたがって平均・分散を取る正規化レイヤー
- ミニバッチのサイズが小さくても、サンプルが1つでも問題ない
- RNNやTransformerで使われる
- 参考: https://data-analytics.fun/2020/07/16/understanding-layer-normalization/

<br>

■Instance Normalization: The Missing Ingredient for Fast Stylization (2016): 各チャンネル独立に画像の縦横方向についてのみ平均・分散を取る正規化レイヤー
- Batch Normalizationのバッチサイズが1の場合、Instance Normalizationと等価
- 参考: https://blog.albert2005.co.jp/2018/09/05/group_normalization/

<br>

■Group normalization (2018): チャンネルを G 個にグルーピングして Layer Normalization と Instance Normalization の中間的な処理を行うレイヤー
- 幅広いバッチサイズでBNよりも安定
- グループの数がハイパーパラメータになる
- 参考: https://blog.albert2005.co.jp/2018/09/05/group_normalization/

<br>

![image](https://qiita-user-contents.imgix.net/https%3A%2F%2Fqiita-image-store.s3.ap-northeast-1.amazonaws.com%2F0%2F730426%2Fb6a0f259-fc33-c54f-30bd-fdf368ff6aae.png?ixlib=rb-1.2.2&auto=format&gif-q=60&q=75&s=4f52b6c4e40890f596afb007168e2a8d)
-------------------------------------------------------
■Weight normalization (2016): ニューラルネット（CNNや全結合層）の重みを正規化するレイヤー
- BNより処理速度早い（BNより計算コスト軽い。CNNは、重みの数は入力の数よりもはるかに少ない傾向があるため）
- WNは重みを再パラメータ化するだけなので、やろうと思えばBNと併用できる
- 全結合層だけのネットワークときにBNの代わりに使えそう
- 参考: 
    - https://www.slideshare.net/KeigoNishida/layer-normalizationnips
    - https://github.com/arXivTimes/arXivTimes/issues/135
    - https://mlexplained.com/2018/01/13/weight-normalization-and-layer-normalization-explained-normalization-in-deep-learning-part-2/

<br>

■Weight Standardization (2019): Weight normalizationの改良。出力チャンネルごとにニューラルネット（CNN）の重みを正規化するレイヤー
- WS + GNとWS + BNの組み合わせでより良い結果を達成したらしい
- 参考: 
    - https://github.com/arXivTimes/arXivTimes/issues/1158
    - https://medium.com/@kushajreal/weight-standardization-a-new-normalization-in-town-54b2088ce355

-------------------------------------------------------
■Adaptive Instance Normalization (2017): 画風変換で使われる正規化レイヤー。変換元の画像の特徴量mapの平均・分散をstyle画像の特徴量mapの平均・分散に合わせる
- AdaINはstyle画像の特徴量mapの平均・分散に合わせるので学習に用いていないstyle画像を使っても画風変換できる
- StyleGANでもAdaIN使われてる
- 参考: 
    - https://medium.com/@crosssceneofwindff/adain%E3%82%92%E7%94%A8%E3%81%84%E3%81%9F%E7%94%BB%E9%A2%A8%E5%A4%89%E6%8F%9B-bcfc2dafbce
    - https://www.slideshare.net/KentoDoi/stylegan-cvpr2019dena

![AdaIN](https://qiita-user-contents.imgix.net/https%3A%2F%2Fqiita-image-store.s3.ap-northeast-1.amazonaws.com%2F0%2F685596%2F95a8c0ab-c4cd-3b8d-5725-7e91b1bdff5f.png?ixlib=rb-1.2.2&auto=format&gif-q=60&q=75&s=a5cc8c09cbf7f9c592e591de3039c0b2)

<br>

■SPADE (2019): AdaINの改良。セマンティックセグメンテーションで使われる。セマンティックマップ画像に対応した画像を生成できる
- 参考: https://qiita.com/Phoeboooo/items/ad6c0461ab052aae8e89

![SPADE](https://qiita-user-contents.imgix.net/https%3A%2F%2Fqiita-image-store.s3.amazonaws.com%2F0%2F315063%2F93fe7ff8-18d8-de1b-76d1-d22bf1841924.png?ixlib=rb-1.2.2&auto=format&gif-q=60&q=75&s=67c09be60736a2906f06a2ff9ed27b78)
